File: app/src/main/java/com/rakib/foodbankbd/OfferActivity.java
package com.rakib.foodbankbd;

import androidx.appcompat.app.AppCompatActivity;

import android.content.pm.ActivityInfo;
import android.graphics.Color;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import com.rakib.foodbankbd.entities.OfferObject;

public class OfferActivity extends AppCompatActivity implements View.OnClickListener {
    private ImageView offerImage;
    private TextView offerName, offerLocation, offerPrice, offerDuration, offerItem, offerDescription;

    private Gson gson;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_offer);

        setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_PORTRAIT);

        offerImage = findViewById(R.id.imageViewOfferImage);
        offerName = findViewById(R.id.textViewOfferName);
        offerLocation = findViewById(R.id.textViewOfferLocation);
        offerPrice = findViewById(R.id.textViewOfferPrice);
        offerDuration = findViewById(R.id.textViewOfferDuration);
        offerItem = findViewById(R.id.textViewOfferItem);
        offerDescription = findViewById(R.id.textViewOfferDescription);

        findViewById(R.id.buttonMap).setOnClickListener(this);

        GsonBuilder gsonBuilder = new GsonBuilder();
        gson = gsonBuilder.create();

        String offerInStringFormat = getIntent().getExtras().getString("OFFER_CODE");
        final OfferObject singleOffer = gson.fromJson(offerInStringFormat, OfferObject.class);

        if (singleOffer != null) {
            offerImage.setImageResource(singleOffer.getOfferImage());
            offerName.append(singleOffer.getOfferName());
            offerLocation.append(singleOffer.getOfferLocation());
            offerPrice.append(Double.toString(singleOffer.getOfferPrice()));
            offerDuration.append(singleOffer.getOfferDuration());
            offerItem.append(singleOffer.getOfferItem());
            offerDescription.append(singleOffer.getOfferDescription());
        }
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.buttonMap:
                Toast.makeText(OfferActivity.this, "This feature is coming soon!", Toast.LENGTH_LONG).show();
                break;
        }
    }
}

File: app/src/main/java/com/rakib/foodbankbd/OfferActivity.kt
package com.rakib.foodbankbd

import android.os.Bundle
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.google.gson.Gson
import com.google.gson.GsonBuilder
import com.rakib.foodbankbd.entities.OfferObject
import kotlinx.android.synthetic.main.activity_offer.*

class OfferActivity : AppCompatActivity(), View.OnClickListener {
    private lateinit var gson: Gson

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_offer)

        val gsonBuilder = GsonBuilder()
        gson = gsonBuilder.create()
        val offerInStringFormat = intent.extras?.getString("OFFER_CODE")
        val singleOffer = gson.fromJson(offerInStringFormat, OfferObject::class.java)

        if (singleOffer != null) {
            imageViewOfferImage.setImageResource(singleOffer.offerImage)
            textViewOfferName.append(singleOffer.offerName)
            textViewOfferLocation.append(singleOffer.offerLocation)
            textViewOfferPrice.append(singleOffer.offerPrice.toString())
            textViewOfferDuration.append(singleOffer.offerDuration)
            textViewOfferItem.append(singleOffer.offerItem)
            textViewOfferDescription.append(singleOffer.offerDescription)
        }
    }

    override fun onClick(v: View) {
        when (v.id) {
            R.id.textViewOfferLocation -> Toast.makeText(this@OfferActivity, "This feature is coming soon!", Toast.LENGTH_LONG).show()
        }
    }
}